examples/multi-decade-example.py

```diff
@@ -0,0 +1,364 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Feb 17 15:25:05 2021
+
+example script to illustrate new features of the multi-decade investment
+
+@author: Lisa
+"""
+import pypsa
+import pandas as pd
+import numpy as np
+
+
+n = pypsa.Network()
+
+# snapshot weightings are now seperated between weightings for
+# (i) the objective function (objective)
+# (ii) elapsed time in the filling of Store and StorageUnit components (stores)
+# (ii) for the contribution of generators to GlobalConstraint components for e.g. CO2 budgets (generators)
+print(n.snapshot_weightings)
+
+# there is a new component investment period weightings
+# split  by weighting for the objective function (objective)
+# and weightings for the elapsed time between the investment periods (time)
+# to calculate CO2 emissions or assets lifetime
+# default is an empty pd.DataFrame
+print(n.investment_period_weightings)
+
+# default snapshots are still single Index
+print(n.snapshots)
+
+# when a pypsa network from an older version is imported, the old snapshot weightings
+# are set row-wise and a warning is given
+n = pypsa.examples.storage_hvdc()
+print(n.snapshot_weightings)
+
+# snapshots can be set as a multiindex
+# a convenience functionality is provided: when setting the investment periods
+# directly, the all time-series are repeated for each period.
+single_snapshots = n.snapshots
+n.investment_periods = [2020, 2030]
+print(n.snapshots)
+print(n.investment_period_weightings)
+assert all(n.snapshots == pd.MultiIndex.from_product([[2020, 2030], single_snapshots]))
+
+# investment_periods have to be integer and increasing, otherwise there will be an
+# error message
+snapshots_multiindex = pd.MultiIndex.from_product([[2030, 2020], single_snapshots])
+n.set_snapshots(snapshots_multiindex)
```

Comment:
This does not fail currently.

Reply:
Yes, we still have to decide on the discussion in https://github.com/PyPSA/PyPSA/pull/259